pkg/tarball/write.go

```diff
@@ -94,6 +129,21 @@ func (ctx *Context) writeTar(tw *tar.Writer, fsys fs.FS) error {
 			header.Gname = ctx.OverrideGname
 		}
 
+		if !info.IsDir() && hasHardlinks(info) {
+			inode, err := getInodeFromFileInfo(info)
+			if err != nil {
+				return err
+			}
```

Comment:
So the first occurrence of an inode we see in the `fs.WalkDir` becomes the source of truth and all subsequent visits become hard links to it, and this is made deterministic by `fs.WalkDir` being deterministic, right?

Reply:
Seems good to me, WalkDir [depends on the underlying FS implementation](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/io/fs/walk.go;l=74-91;drc=refs%2Ftags%2Fgo1.18) if we notice inconsistencies we can override ReadDir in apko/fs and make the return order deterministic.

Reply:
yes